File: 프로그래머스/lv1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.java
import java.util.*;
class Solution {
    public String[] solution(int n, int[] arr1, int[] arr2) {
        String[] answer = new String[n];
		String[] map1 = getMap(n, arr1);
		String[] map2 = getMap(n, arr2);

		Arrays.fill(answer, "");

		for(int i = 0; i < n; i++) {
			for(int j = 0; j < n; j++) {
				if(map1[i].charAt(j) == '1' || map2[i].charAt(j) == '1') {
					answer[i] += "#";
				} else {
					answer[i] += " ";
				}
			}
		}
        return answer;
    }
    
    public String[] getMap(int n, int[] arr) {
		String[] map = new String[n]; 

		for(int i = 0; i < n; i++) {
			map[i] = Integer.toBinaryString(arr[i]);
			while(map[i].length() < n) {
				map[i] = "0" + map[i];
			}
		}
		return map;
	}
}

File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.js
function solution(n, arr1, arr2) {
    var answer = [];
    arr1 = arr1.map(v => v.toString(2).padStart(arr1.length, '0'));
    arr2 = arr2.map(v => v.toString(2).padStart(arr2.length, '0'));    
    for (let i = 0; i < arr1.length; i++){
        let row = '';
        for (let j = 0; j < arr1[i].length; j++){
            if (arr1[i][j] ==='1' || arr2[i][j] === '1') row += '#';
            else row += ' ';
        }
        answer.push(row);
    }
    return answer;
}

